app/providers.tsx

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import * as Sentry from "@sentry/nextjs";
+import posthog from "posthog-js";
+import { PostHogProvider } from "posthog-js/react";
+import { useEffect } from "react";
+
+export function Analytics({
+	children,
+}: {
+	children: React.ReactNode;
+}) {
+	if (process.env.NEXT_PUBLIC_ENABLE_ANALYTICS !== "true") {
+		return <>{children}</>;
+	}
```

Comment:
This code runs before the useEffect, but after the full function component body. This means even when analytics are disabled, the component will still try to use the PostHogProvider, which could cause errors since posthog won't be initialized.

https://react.dev/reference/rules/rules-of-hooks

> [!NOTE]
> I was expecting an error with biome, but was surprised that it didn't 🤔 

A better structure would be:

```tsx
export function Analytics({ children }: { children: React.ReactNode }) {
    if (process.env.NEXT_PUBLIC_ENABLE_ANALYTICS !== "true") {
        return <>{children}</>;
    }

    return (
        <AnalyticsEnabled>
            {children}
        </AnalyticsEnabled>
    );
}

function AnalyticsEnabled({ children }: { children: React.ReactNode }) {
    useEffect(() => {
        try {
            // ... initialization code ...
        } catch (error) {
            // ... error handling ...
        }
    }, []);

    return <PostHogProvider client={posthog}>{children}</PostHogProvider>;
}
```

Reply:
Thank you. Fixed https://github.com/giselles-ai/giselle/pull/112/commits/a963775105af8f4b444ee24515f9549dd1531843